docs/rfd/001.md

```diff
@@ -0,0 +1,181 @@
+*   The protocol gives advice to "maximize the human readability of the URL", which conflicts with an unambiguous presentation of said characters in percent-encoded form.
+
+
+# Requirements
+
+*   There must be a well-defined interpretation of the GLOME Login handshake that does not depend on the state of the server.
+*   There must be a well-defined, bijective conversion from the message embedded in a GLOME Login URL to the message being authorized.
+*   Subject to the preceding requirements, the URL layout should be optimized for human readability (e.g. don't encode unreserved characters, go/rfc/3986#section-2.3) and brevity.
+    -   Assuming humans will have to read the message to be authorized much more often than parse the involved keys.
+
+
+# Design ideas
+
+*   Make the URL generation and interpretation unambiguous.
+*   Include detailed instructions for server and client into the protocol.
+*   Publish the result as GLOME Login v2.
+
+
+## URL Format
+
+```
+<url-prefix>/v2/<server-pub-trunc>/<client-pub>/<message>/
+```
+
+## Handshake
+
+The handshake is simplified to consist of a (truncated) public key of the server and the full public key of the client, given in separate URL segments. The length of the server key should be chosen so that the likelihood of generating conflicting keys is low (see Appendix).
+
+Both public keys are converted to URI segments using URL-safe base64 encoding.
+
+
+## Message
+
+Define a normalized encoding of the authorization context and compute the tag over that encoding.
+
+New restrictions:
+
+*   hostid-type and hostid must not contain the `:` character.
+*   hostid and hostid-type should not contain any characters that would be escaped in a URI path segment (as detailed below). Differing from previous protocol versions, `/` is discouraged.
+
+
+### URI Path Segments
+
+The URI specification [RFC 3986](go/rfc/3986#section-3.3)[^1] defines a path segment as
+
+```
+segment = *pchar
+pchar = unreserved / pct-encoded / sub-delims / ":" / "@"
+unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
+sub-delims  = "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "="
+pct-encoded = "%" HEXDIG HEXDIG
+```
+
+where HEXDIG should refer to a digit or an uppercase letter A-F. This matches the definition in https://url.spec.whatwg.org/#url-path-segment-string, which supposedly supersedes the RFC.
+
+Thus, define `EscapePathSegment` as a function that escapes all characters that are not unreserved, sub delimiters, `:` or `@`. See the Appendix for how this function can be implemented in some of the major programming languages.
+
+
+### Message Encoding
+
+Constructing a message takes three parameters: hostid, action and (optionally) hostid-type. The content to be embedded in the URL consists of 2 path segments: host-segment and action-segment.
```

Comment:
What would be the benefits of restricting the action only to one URL path segment? To detect that the URL has not been truncated (provided that there is a trailing slash at the end)?

Reply:
@l9i please have a look at my argument above.